cxotime/cxotime.py

```diff
@@ -498,3 +502,40 @@ def to_value(self, parent=None, **kwargs):
         return out
 
     value = property(to_value)
+
+
+class CxoTimeDescriptor(TypedDescriptor):
+    """Descriptor for an attribute that is CxoTime (in date format) or None if not set.
+
+    This allows setting the attribute with any ``CxoTimeLike`` value.
+
+    Note that setting this descriptor to ``None`` will set the attribute to ``None``,
+    which is different than ``CxoTime(None)`` which returns the current time. To set
+    an attribute to the current time, set it with ``CxoTime.now()``.
+
```

Comment:
Not a lien, but I could use an example of how to set a dataclass attribute to the current time with CxoTime.now().

Reply:
In the current version you cannot do this in the class definition as a default value. I.e. there is no way to set a `stop` attribute to default to NOW if it is not set. It has to be done in the object instantiation. And that is what I showed in the comment above.

I might find a way, but it will involve adding a sentinel value to `CxoTime` and changing the `TypedDescriptor` to be lazy about using the default value (which I tried and failed to do once).

Reply:
And by "I could use", did you mean in the doc string or as a comment here in the PR?

Reply:
I thought your comment that there was no way to set the `stop` to a CxoTimeDescriptor with a default value was different from initializing an attribute of the dataclass to CxoTime.now(), but I wasn't sure how to do that either.

Reply:
```
@dataclass
class MyClass:
    stop: CxoTime | None = CxoTimeDescriptor()

obj = MyClass(stop=CxoTime.now())
```

Reply:
https://github.com/sot/cxotime/pull/43#issuecomment-1878862961